### PR TITLE
implemented notifications

### PR DESCRIPTION
**Pull request by resoluteCoder**
Jira ticket: https://issues.redhat.com/browse/AAP-8174

Before:
image

After:
image

### CODE REVIEW

File: frontend/awx/resources/notifications/hooks/useNotificationColumns.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      });
+    },
+    [pageNavigate]
+  );
+  const nameColumn = useNameColumn({ ...options, onClick: nameClick });
+  const typeColumn = useMemo<ITableColumn<NotificationTemplate>>(
+    () => ({
+      header: t('Type'),
+      type: 'text',
+      value: (notificationTemplate: NotificationTemplate) => {
```
Comment (matoval) [inline frontend/awx/resources/notifications/hooks/useNotificationColumns.tsx:29]: Types should be capitalized

File: frontend/awx/resources/notifications/hooks/useNotificationColumns.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        return notificationTemplate.notification_type;
+      },
+      card: 'subtitle',
+      list: 'subtitle',
+    }),
+    [t]
+  );
+  const tableColumns = useMemo<ITableColumn<NotificationTemplate>[]>(
```
Comment (matoval) [inline frontend/awx/resources/notifications/hooks/useNotificationColumns.tsx:37]: Do we want a header of Options above the start, success, failure toggles?

**Review by matoval — COMMENTED** [review]
Looks great, I left a few commands about a few tiny details:

The breadcrumb for sources is missing a link for the sources name(sources/:source_id)

File: frontend/awx/resources/notifications/hooks/useNotificationColumns.tsx (hunk 2 above)
Comment (resoluteCoder) [reply frontend/awx/resources/notifications/hooks/useNotificationColumns.tsx:37]: I'm not entirely sure. The old UI does have them but I wasn't able to figure out how to use the switch toggle within a column. It seems that the new UI uses the switch toggle component in the action part of the row. I could be wrong of course :sweat_smile:

File: frontend/awx/resources/notifications/hooks/useNotificationColumns.tsx (hunk 1 above)
Comment (resoluteCoder) [reply frontend/awx/resources/notifications/hooks/useNotificationColumns.tsx:29]: Not sure what you mean here. However I did fix where I had `Notification Type` where it seems the rest of the UI uses capital then lower `Notification type`

**Review by matoval — APPROVED** [review]

**Pull request merged**